StudentDriver/StudentDriver/Services/WebService.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using StudentDriver.Models;
+using Xamarin.Forms;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using StudentDriver.Helpers;
+
+namespace StudentDriver.Services
+{
+    public static class WebService
+    {
+        private static string BaseUrl = "";
+
+        public static async Task<string> GetAuthentiationToken(OAuthProvider.ProviderType providerType, string OAuthId)
+        {
+            var client = new HttpClient();
+            var requestUri = GenerateRequestUri(BaseUrl, "authenticate", new Dictionary<string, string>() { { "providerType", providerType.ToString() }, {"id",OAuthId} });
+            var response = await client.GetAsync(requestUri);
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                return null;
+            }
+            var json = response.Content.ReadAsStringAsync().Result;
+            var token = JsonConvert.DeserializeObject<string>(json);
+            return token;
+        }
+
+
+        public static async Task<UserStats> GetStudentStats(int id, string token)
+        {
+            var client = new HttpClient();
+            var requestUri = GenerateRequestUri(BaseUrl,"students", new Dictionary<string, string>() {{"userId", id.ToString()}, {"token", token} });
+            var response = await client.GetAsync(requestUri);
+            var json = response.Content.ReadAsStringAsync().Result;
+            var userStats = JsonConvert.DeserializeObject<UserStats>(json);
+            return userStats;
+        }
+
+        public static async Task<bool> PostUnsyncDrivingSessions(List<UnsyncDrive> unsyncDrives, string token)
+        {
+            var json = new JObject(new JProperty("token", token),"unsyncDrives",unsyncDrives);
+
+            var content = new StringContent(json.ToString(), Encoding.UTF8, "application/json");
+            var client = new HttpClient();
+            var requestUri = GenerateRequestUri(BaseUrl,"drivingsessions");
+
+            var response = await client.PostAsync(requestUri, content);
+            return response.IsSuccessStatusCode;
+        }
```

Comment:
@happy25 Will there be a POST for a list of DrivePoints for a DriveSession? From a discussion with @zawadzkip, we were thinking that mobile will send a list of points to the web server for a drive session and the server will store the session after processing the points. If so, what would the JSON data look like for the request? (This will help me with #20) :D